### PR TITLE
fixed mapbase autoalign

### PR DESCRIPTION
**Pull request by Mihan786Chistie**
Fixed the issue: https://github.com/sunpy/sunpy/issues/6794

### CODE REVIEW

**Comment by Mihan786Chistie** [comment]
pre-commit.ci autofix

**Review by Mihan786Chistie — COMMENTED** [review]
Changed `ret = axes.pcolormesh(data, **imshow_args)` to
`ret = axes.pcolormesh(np.arange(data.shape[1] + 1) - 0.5,
                                  np.arange(data.shape[0] + 1) - 0.5,
                                  data, **imshow_args)`

**Comment by Mihan786Chistie** [comment]
Can @ayshih review the changes made are ok?

**Comment by nabobalis** [comment]
Hi @Mihan786Chistie, thanks for the pull request.

A few things need to occur with this bugfix. 

1. You will need to add a bugfix changelog entry following these instructions: https://github.com/sunpy/sunpy/blob/main/changelog/README.rst
2. This change has changed the figure hashes used by ci/circleci. You can see the output here: https://app.circleci.com/pipelines/github/sunpy/sunpy/20284/workflows/bbaff9b2-a494-4580-9d13-ccf9d50902c5/jobs/83621

This needs to be changed for both files we have: https://github.com/sunpy/sunpy/blob/main/sunpy/tests/figure_hashes_mpl_352_ft_261_astropy_51_animators_100.json and https://github.com/sunpy/sunpy/blob/main/sunpy/tests/figure_hashes_mpl_dev_ft_261_astropy_dev_animators_dev.json

The docbuild failure you can ignore, that was my fault.

**Comment by Mihan786Chistie** [comment]
I think I understood how to add changelog, but I am confused how to fix the ci/circleci test an what to change in both the file https://github.com/sunpy/sunpy/blob/main/sunpy/tests/figure_hashes_mpl_352_ft_261_astropy_51_animators_100.json and https://github.com/sunpy/sunpy/blob/main/sunpy/tests/figure_hashes_mpl_dev_ft_261_astropy_dev_animators_dev.json. Can you explain bit of the process?

**Comment by Mihan786Chistie** [comment]
@nabobalis please review this PR

**Comment by nabobalis** [comment]
> @nabobalis please review this PR

Please do not @ people unless it is urgent.

File: changelog/6796.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fixed a bug in ### sunpy/map/mapbase.py file to supply the pixeledges to the ####pcolormesh() function explicitly while calling pcolormesh() function, so it does not treat integer pixels as pixel edges.
```
Comment (nabobalis) [inline changelog/6796.bugfix.rst:1]:
```suggestion
When using ``autoalign=True`` when plotting maps, the result was off by half a pixel as the ``pcolormesh`` call was not supplying the pixel edges explicitly.
```

**Review by ayshih — CHANGES_REQUESTED** [review]
Please add the following new figure test to the bottom of the file `sunpy/map/tests/test_plotting.py`:
```python
@figure_test
def test_plot_autoalign_pixel_alignment(aia171_test_map):
    # Verify that autoalign=True does not affect pixel alignment
    x, y = [z.value for z in aia171_test_map.reference_pixel]

    fig = Figure(figsize=(10, 4))

    ax1 = fig.add_subplot(121, projection=aia171_test_map)
    aia171_test_map.plot(axes=ax1, autoalign=False, title='autoalign=False')
    ax1.grid(False)
    ax1.set_xlim(x - 2, x + 2)
    ax1.set_ylim(y - 2, y + 2)

    ax2 = fig.add_subplot(122, projection=aia171_test_map)
    aia171_test_map.plot(axes=ax2, autoalign=True, title='autoalign=True')
    ax2.grid(False)
    ax2.set_xlim(x - 2, x + 2)
    ax2.set_ylim(y - 2, y + 2)

    return fig
```

The current, bugged code produces (the panels are supposed to match):
Figure_1

**Comment by Mihan786Chistie** [comment]
pre-commit.ci autofix

File: sunpy/map/tests/test_plotting.py [hunk 2]
```diff
@@ -345,3 +345,25 @@ def test_plot_autoalign(aia171_test_map):
 def test_plot_autoalign_bad_inputs(aia171_test_map):
     with pytest.raises(ValueError):
         aia171_test_map.plot(autoalign='bad')
+
+
+@figure_test
+def test_plot_autoalign_pixel_alignment(aia171_test_map):
```
Comment (nabobalis) [inline sunpy/map/tests/test_plotting.py:351]: You will need to add this hash to the two figure hash files. You can get the hash and test name from the failed CircleCi runs.

File: changelog/6796.bugfix.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+When using ``autoalign=True`` when plotting maps, the result was off by half a pixel as the ``pcolormesh`` call was not supplying the pixel edges explicitly.
```
Comment (ayshih) [inline changelog/6796.bugfix.rst:1]:
```suggestion
When using ``autoalign=True`` when plotting maps, the result was misaligned by half a pixel.
```
There's no particular benefit here to providing detail

**Review by ayshih — APPROVED** [review]
Just one suggestion on the changelog entry, otherwise looks good to me

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Thanks @Mihan786Chistie for the PR and @ayshih for the review.

**Comment by Mihan786Chistie** [comment]
Thanks @nabobalis and @ayshih for guiding me through and approving my PR.

**Pull request merged**